app/api/__tests__/get-random-whys.js

```diff
@@ -0,0 +1,242 @@
+// https://github.com/EnCiv/civil-pursuit/issues/135
+
+import getRandomWhys from '../get-random-whys'
+import Points from '../../models/points'
+import { Mongo } from '@enciv/mongo-collections'
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { ObjectId } from 'mongodb'
+
+const USER1 = '6667d5a33da5d19ddc304a6b'
+const POINT1 = new ObjectId('6667d688b20d8e339ca50020')
+let MemoryServer
+
+beforeAll(async () => {
+  MemoryServer = await MongoMemoryServer.create()
+  const uri = MemoryServer.getUri()
+  await Mongo.connect(uri)
+})
+
+afterAll(async () => {
+  await Mongo.disconnect()
+  await MemoryServer.stop()
+})
+
+test('Fail if user is not logged in', async () => {
+  const cb = jest.fn()
+
+  await getRandomWhys.call({}, POINT1.toString(), 'most', 5, cb)
+
+  expect(cb).toHaveBeenCalledTimes(1)
+  expect(cb).toHaveBeenCalledWith(undefined)
+})
+
+test("Fail if parameters missing or don't match", async () => {
+  const user = { id: USER1 }
+  const cb = jest.fn()
+
+  await getRandomWhys.call({ synuser: user }, null, 'most', 5, cb)
+  expect(cb).toHaveBeenCalledTimes(1)
+  expect(cb).toHaveBeenCalledWith(undefined)
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), null, 5, cb)
+  expect(cb).toHaveBeenCalledTimes(2)
+  expect(cb).toHaveBeenCalledWith(undefined)
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), 'most', null, cb)
+  expect(cb).toHaveBeenCalledTimes(3)
+  expect(cb).toHaveBeenCalledWith(undefined)
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), 'most', 'five', cb)
+  expect(cb).toHaveBeenCalledTimes(4)
+  expect(cb).toHaveBeenCalledWith(undefined)
+})
+
+test('Case where there are 0 matching whys', async () => {
+  const user = { id: USER1 }
+  const cb = jest.fn()
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), 'least', 5, cb)
+
+  expect(cb).toHaveBeenCalledTimes(1)
+  const result = cb.mock.calls[0][0]
+  expect(result).toEqual([])
+})
+
+test('Case where there are 5 matching whys and qty is 5', async () => {
+  const whys = [
+    {
+      _id: new ObjectId(),
+      title: 'Why 1',
+      description: 'Description 1',
+      round: 1,
+      parentId: POINT1,
```

Comment:
I just realized I need to point out that parentId, like userId should be a string and not an ObjectId.  I didn't specify this I need to add it to documentation somewhere.  

There reason is that it's hard to pass ObjectIds between browsers and server - they get converted to strings, and it becomes confusing and difficult to deal with in code on the server and browser side.  So my convention is that all xxxxId's are strings.  _id will be an ObjectId on the server side, and string on the browser side. 

Please look back at upserts and see if any changes need to be made. 

Also, the apis should check the _id of the object coming in, and if it's present, convert it from a string to an ObjectId.